instrumentation/okhttp/okhttp-3.0/library/src/main/java/io/opentelemetry/instrumentation/okhttp/v3_0/OkHttpTelemetryBuilder.java

```diff
@@ -62,13 +65,35 @@ public OkHttpTelemetryBuilder setCapturedResponseHeaders(List<String> responseHe
     return this;
   }
 
+  /**
+   * Configures the instrumentation to recognize an alternative set of HTTP request methods.
+   *
+   * <p>By default, this instrumentation defines "known" methods as the ones listed in <a
+   * href="https://www.rfc-editor.org/rfc/rfc9110.html#name-methods">RFC9110</a> and the PATCH
+   * method defined in <a href="https://www.rfc-editor.org/rfc/rfc5789.html">RFC5789</a>.
+   *
+   * <p>Note: calling this method <b>overrides</b> the default known method sets completely; it does
+   * not supplement it.
+   *
+   * @param knownMethods A set of recognized HTTP request methods.
+   */
+  @CanIgnoreReturnValue
+  public OkHttpTelemetryBuilder setKnownMethods(Set<String> knownMethods) {
```

Comment:
`@Nullable` on the param?

Reply:
Collections should never be nullable in the API (see  [the comment above](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/8779#discussion_r1241950700))